### DISCUSSION TITLE
"Rendered by React server-side."?

### DISCUSSION THREAD

**Discussion by pbowyer** [silviogutierrez/reactivated#162] [Q&A] [answered]
The homepage says "Rendered by React server-side." but reading the docs it looks like the Django templates are rendered server-side and the React is served as JS and gets its props from the Django template.

Am I way off, or does reactivated do server-side rendering of the React components?

**Comment by silviogutierrez** [comment, marked as answer]
So everything is rendered by the server. There are no Django templates involved. Or rather, your templates are just React components.

See the home page of the docs:

https://github.com/silviogutierrez/reactivated/blob/main/website/client/templates/HomePage.tsx

And now got to the site at https://www.reactivated.io. View the source or turn off JS. Notice it's fully rendered and no JS is required.

Of course, if you want dynamic behavior, like `onClick` handlers or so, you can use them. React will hydrates on the client and enable any client-only behaviors, *without* re-rendering.

Hope that clarifies.

**Reply by pbowyer** [reply]
Ooooh that's magic!

**Reply by silviogutierrez** [reply]
Just about!
